packages/datagateway-dataview/src/App.tsx

```diff
@@ -106,6 +106,27 @@ const queryClient = new QueryClient({
   },
 });
 
+export const QueryClientSettingUpdater: React.FC<{
+  queryClient: QueryClient;
+}> = (props) => {
+  const { queryClient } = props;
+  const queryRetries = useSelector(
+    (state: StateType) => state.dgcommon.queryRetries
+  );
+
+  React.useEffect(() => {
+    if (typeof queryRetries !== 'undefined') {
+      const opts = queryClient.getDefaultOptions();
+      queryClient.setDefaultOptions({
+        ...opts,
+        queries: { ...opts.queries, retry: queryRetries },
+      });
+    }
+  }, [queryClient, queryRetries]);
+
+  return null;
+};
```

Comment:
Could this be put in common instead of replicating it in each package?

Reply:
Technically I can extract the search and dataview ones as they're the same, but download is different due to not storing the settings in redux. At that point I wondered if it's just easier to keep each one defined explicitly where it's used, but happy to extract out search & dataview's implementation to common if you think it makes things tidier

Reply:
Ah right, I didn't see that. I think it would be nice to at least have the identical ones in common. Not sure what you mean by the second option.

Reply:
It might be cleaner to extract the shared functionality into a common component and pass the queryRetries as a prop, keeping Redux and Context-specific logic separate.

Then it can used like this in the relvant App.tsx 
```
export const QueryClientSettingUpdaterContext: React.FC<{
  queryClient: QueryClient;
}> = ({ queryClient }) => {
  const { queryRetries } = useContext(DownloadSettingsContext);

  return (
    <QueryClientSettingUpdaterCommon
      queryClient={queryClient}
      queryRetries={queryRetries}
    />
  );
};
```

Reply:
@joshuadkitenge yep that was my current plan, and maybe even have the hook up to redux in common so search & dataview can have the same one but download can still share most of the code